### PR TITLE
Disable gcc4.4 C++ tests - DO NOT MERGE INTO V1.0.X

### PR DESCRIPTION
**Pull request by matt-kwong**
Since we will be dropping support for non-C++11 compilers, unsupported compiler tests should be disabled.

### CODE REVIEW

**Comment by ctiller** [comment]
Can we also remove gcc 4.6, and instead test 4.8.

**Comment by vjpai** [comment]
Agreed with @ctiller - we need to remove gcc 4.6 as well because it also lacks key C++11 features.

File: tools/run_tests/run_tests_matrix.py [hunk 1]
```diff
@@ -181,7 +181,17 @@ def _create_portability_test_jobs(extra_args=[], inner_jobs=_DEFAULT_INNER_JOBS)
   # portability C and C++ on x64
   for compiler in ['gcc4.4', 'gcc4.6', 'gcc5.3',
```
Comment (dgquintas) [inline tools/run_tests/run_tests_matrix.py]: shouldn't something change in this line as well?

Comment (matt-kwong) [inline tools/run_tests/run_tests_matrix.py]: After talking with Craig, we thought it wouldn't make a difference if we changed from 4.6 to 4.8 for C tests, so I decided to not change it. I would be open to changing it if it was better for testing though.

**Comment by vjpai** [comment]
Sanity tests failing right now; otherwise, largely seems to be known flakes.

**Comment by vjpai** [comment]
gcc-4.8 is not found. I think the Dockerfile needs changing.

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -332,6 +332,8 @@ def _compiler_options(self, use_docker, compiler):
       return ('wheezy', self._gcc_make_options(version_suffix='-4.4'))
     elif compiler == 'gcc4.6':
       return ('wheezy', self._gcc_make_options(version_suffix='-4.6'))
+    elif compiler == 'gcc4.8':
+      return ('wheezy', self._gcc_make_options(version_suffix='-4.8'))
```
Comment (vjpai) [inline tools/run_tests/run_tests.py]: I'm wondering if wheezy is the right choice here or if this should be based on jessie instead, in the overall spirit of choosing newer platforms for C++. In either case, I think you'll need to apt-get gcc-4.8/etc in the Dockerfile template.

**Comment by matt-kwong** [comment]
Definitely not using Wheezy - I see your point and have had issues with Wheezy in the past and want to avoid adding more dependency on it. I tried using Ubuntu1604, which we use for 5.3, and it seems to work.

Also, are there any strong opinions on adding gcc4.8 as one of the compilers to test on for C?

**Comment by vjpai** [comment]
I have no strong feelings about C on 4.8. I'm ok with this. In fact, I would suggest that it's better if this PR just focuses on changing the C++ side, where 4.8 is a relevant milestone. This PR can leave C alone, IMO.

File: templates/tools/dockerfile/test/cxx_ubuntu1604_x64/Dockerfile.template [hunk 3]
```diff
@@ -39,7 +39,7 @@
   # The clang-3.6 symlink for the default clang version was added
   # to Ubuntu 16.04 recently, so make sure it's installed.
   # Also install clang3.7.
-  RUN apt-get update && apt-get -y install clang-3.6 clang-3.7 && apt-get clean
+  RUN apt-get update && apt-get -y install gcc-4.8 g++-4.8 clang-3.6 clang-3.7 && apt-get clean
```
Comment (vjpai) [inline templates/tools/dockerfile/test/cxx_ubuntu1604_x64/Dockerfile.template]: I think we also need gcc-4.8-multilib and g++-4.8-multilib for this to work, or at least to be consistent with everything else

**Comment by matt-kwong** [comment]
Portability tests are passing; I'll keep an eye on the master portability to see if there are any issues after merging.

**Pull request merged**